added codim one hyper ball

nicolacavallini opened the pull request:
I added codimension one for 
hyper_ball<1,2> and hyper_ball<2,3>

bangerth commented on include/deal.II/grid/grid_generator.h:
```diff
@@ -358,6 +358,19 @@ namespace GridGenerator
+  template <int dim, int spacedim>
+  void hyper_ball (Triangulation<dim,spacedim> &tria,
+                   const Point<spacedim>   &center = Point<spacedim>(),
+                   const double        radius = 1.);
```
Why don't you make this the only declaration available? In other words, give the original declaration a second template argument `spacedim`?

bangerth commented on source/grid/grid_generator.cc:
```diff
@@ -2861,7 +2861,19 @@ namespace GridGenerator
+    std::set<types::boundary_id> boundary_ids;
+    boundary_ids.insert (0);
+    GridGenerator::extract_boundary_mesh (volume_mesh, tria,
+                                          boundary_ids);
```
uh oh, I now understand. When you say `hyperball<2,3>` you mean the *surface* of the ball, right? I think then you're using the wrong word. In mathematical English, a *ball* are all the points `||x|| <= r` whereas a *sphere* are all the points with `||x||=r`. In other words, what you're creating here is a hypersphere. I think that's how you should name the function.

bangerth commented on tests/grid/grid_generator_hyper_ball_codim_one.cc:
```diff
@@ -0,0 +1,72 @@
+
+
+
+// Test GridGenerator::create_triangulation_with_removed_cells
```
need to adjust this comment

bangerth commented:
Please adjust the name of the function. Do you happen to have pictures of the meshes? If so, we could also embed them into the documentation.

nicolacavallini commented:
Hi!  Thanks for the comments!

Adjusted function name.

Is this the cosset location to place my pics?
dealii/doc/doxygen/images/

I include them with:

@image html pic_name.png

right?

nicolacavallini commented:
guys somebody included bad indentation to:
(this is the git status after running the indentation script)

	modified:   source/base/function_lib.cc
	modified:   tests/base/functions_10.cc

travis is complaining

bangerth commented on include/deal.II/grid/grid_generator.h:
```diff
@@ -358,6 +358,38 @@ namespace GridGenerator
+    * This function is declared to exist for dim=1, spacedim =2, and
+    * dim=2, spacedim =3.
+    *
+    * The following pictures are genreted with:
```
genreted -> generated

bangerth commented on include/deal.II/grid/grid_generator.h:
```diff
@@ -358,6 +358,38 @@ namespace GridGenerator
+    * @endcode
+    *
+    * See the @ref manifold "documentation module on manifolds", for
+    * bettere detail.
```
bettere -> more

bangerth commented:
Thanks for renaming and adding the pictures. Can you please fix the two typos?

nicolacavallini commented:
Typos fixed.

tamiko commented:
Intel is a bit picky about a minor issue [1]. :wink:

[1] http://cdash.kyomu.43-1.org/viewBuildError.php?type=1&onlydeltap&buildid=12589

bangerth commented on source/grid/grid_generator.cc:
```diff
@@ -4317,6 +4331,18 @@ namespace GridGenerator
+  template void
+  hyper_sphere< 1 ,  2 > (Triangulation< 1 ,   2> &,
+                          const Point<2>   &center = Point<2>(),
+                          const double        radius = 1.);
```
Can you come up with a follow up patch that removes the default values of the arguments here and in the next couple of lines to silence the earnings @tamiko observes?
